engine/src/test/java/com/datatorrent/stram/engine/RecoverableInputOperator.java

```diff
@@ -134,4 +129,13 @@ void setSimulateFailure(boolean flag)
   {
     simulateFailure = flag;
   }
+
+  @Override
+  public void beforeCheckpoint(long windowId) {
+    logger.debug("checkpointedWindowId {} checkpointed at {}", Codec.getStringWindowId(checkpointedWindowId), Codec.getStringWindowId(windowId));
```

Comment:
Should it be logged after checkpointedWindowId is changed as the value that will be written to a checkpoint is the modified value? Please consider overriding toSting() similar to LogicalPlan.toString(), so logging can be simplified.